src/librustc_mir/build/expr/as_temp.rs

```diff
@@ -37,6 +37,14 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         let temp = this.temp(expr_ty.clone());
         let temp_lifetime = expr.temp_lifetime;
         let expr_span = expr.span;
+        let source_info = this.source_info(expr_span);
+
+        if temp_lifetime.is_some() {
+            this.cfg.push(block, Statement {
+                source_info: source_info,
+                kind: StatementKind::StorageLive(temp.clone())
```

Comment:
Make a helper function in cfg, since you’re constructing this multiple times manually.

This should become `this.cfg.push_storage_live(source_info, temp);`